cips/cip-x.md

```diff
@@ -0,0 +1,79 @@
+| cip | XX (assigned by Editors) |
+| - | - |
+| title | Header Pruning for Light Nodes |
+| description | Mechanism to retain a fixed range of headers instead of whole history |
+| author | Hlib Kanunnikov ([@Wondertan](https://github.com/Wondertan)) |
+| discussions-to | <https://forum.celestia.org/t/cip-header-pruning-for-lns/1958> |
+| status | Draft |
+| type | Standards Track |
+| category | Data Availability |
+| created | 2025-03-21 |
+
+## Abstract
+
+Currently, every data availability (DA) node type synchronizes all historical headers starting from genesis (or other statically configured
+historical header) until the subjectively initialized head of the chain. We change that by adding a way to sync a
+constant size range of headers instead of the whole history.
+
+## Motivation
+
+Light nodes (LNs) currently store the entire history of ExtendedHeaders, including the ValidatorSet, Commit, and
+Data Availability Header (DAH). This started as technical debt but has now become a product issue. The goal is to
+reduce LN storage requirements by retaining only the most recent headers—specifically, at least the number defined by
+SampleWindow in [CIP-004](./cip-004.md).
+
+## Specification
+
+In order to reference the last stored header, we introduce a notion of **Tail** header of the chain, complementing
+**Head** header. Tail header height is estimated with [HeaderPruningWindow](#parameters), block time and Head height.
+
+- On the first start of a node, the Tail is retrieved from trusted peers alongside with the Head, performing subjective
+initialization.
+  - Once retrieved, Tail becomes the point of trust for the initializing node where everything beforehand assumed valid
+ (unless syncing from genesis)
+  - Head is cross-checked with the Tail and becomes a synchronization target for the node.
+  - All the headers from Tail to Head are then retrieved and verified using signature-based forward verification.
+- As the chain progresses, nodes continuously re-estimate Tail and cut off stored headers beyond the new Tail, retaining
+roughly [HeaderPruningWindow](#parameters) amount of headers.
+
+The estimation of Tail height is done as follows:
+
+```go
+func estimateTail(head Header, blockTime, headerPruningWindow time.Duration) (height uint64) {
+    headersToRetain := headerPruningWindow / blockTime
+    tail := head.Height() - headersToRetain
+    return tail
+}
+```
+
+## Parameters
+
+| Parameter           | Proposed value | Description                                   | Changeable via Governance |
+|---------------------|----------------|-----------------------------------------------|---------------------------|
+| HeaderPruningWindow | 30 days        | Defines the length of time to retain headers  | No                        |
+
+By default, HeaderPruningWindow (HPW) equals to SamplingWindow (SW) from [CIP-004](./cip-004.md). It can be modified per LN, but
```

Comment:
```suggestion
By default, HeaderPruningWindow (HPW) is equal to SamplingWindow (SW) from [CIP-004](./cip-004.md). It can be modified per LN, but
```